components/cmp/services/src/useCases/GetPurposesAndVendorsUseCase.js

```diff
@@ -3,6 +3,31 @@ export class GetPurposesAndVendorsUseCase {
     this._repository = repository
   }
 
+  execute({retrieveConsentsFromCmp} = {}) {
```

Comment:
You're returning promises anyways, so this should be enoough:
```js
execute({retrieveConsentsFromCmp} = {}) {
   return retrieve ? this._loadStoredConsents() : this._loadDefaultConsents()
}
```

Reply:
execute is the method were all starts in this use case, so it starts with a Promise flow.
Also, I don't like the idea of depending on returning values of private methods to know what's the type of the returning value for this method.
Anyway, I understand your point but if this code grows, when you will start to think in using a Promise flow?
;)

Reply:
I don't what you mean with "promise flow". Only if you want to catch errors wrapping everything into a promise.
Either are fine. async funcs too.